app/src/main/java/org/thunderdog/challegram/component/chat/ReactionBubble.java

```diff
@@ -0,0 +1,256 @@
+package org.thunderdog.challegram.component.chat;
+
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.Paint;
+import android.graphics.Path;
+import android.graphics.Rect;
+import android.graphics.RectF;
+import android.view.MotionEvent;
+import android.view.View;
+
+import androidx.annotation.NonNull;
+
+import org.drinkless.td.libcore.telegram.Client;
+import org.drinkless.td.libcore.telegram.TdApi;
+import org.thunderdog.challegram.R;
+import org.thunderdog.challegram.core.Lang;
+import org.thunderdog.challegram.data.TD;
+import org.thunderdog.challegram.data.TGMessage;
+import org.thunderdog.challegram.loader.ImageFile;
+import org.thunderdog.challegram.loader.ImageReceiver;
+import org.thunderdog.challegram.loader.gif.GifFile;
+import org.thunderdog.challegram.telegram.Tdlib;
+import org.thunderdog.challegram.theme.Theme;
+import org.thunderdog.challegram.tool.Paints;
+import org.thunderdog.challegram.tool.Screen;
+import org.thunderdog.challegram.util.text.Text;
+import org.thunderdog.challegram.util.text.TextPart;
+import me.vkryl.core.lambda.Destroyable;
+
+
+public class ReactionBubble extends View implements Client.ResultHandler, Destroyable {
+
+    public ReactionBubble(@NonNull TdApi.MessageReaction reaction, @NonNull Tdlib tdlib, @NonNull TGMessage context, int key) {
+        super(context.context());
+        this.reaction = reaction;
+        this.tdlib = tdlib;
+        this.context = context;
+        this.key = key;
+        text = new Text(getText(), 999, Paints.getTitleStyleProvider(), () -> getTextColor(), 1, 0, null);
+        this.reactionInfo = findReactionInfo(reaction.reaction, tdlib);
+    }
+    private TdApi.MessageReaction reaction;
+    private  TdApi.Reaction reactionInfo;
```

Comment:
I can notice that PR has some issues with whitespaces:

1. Double whitespace before field type
2. Missing whitespace after `for`, `if`, etc
3. Double whitespace before method return type

Since whitespaces fixes are minor and do not affect the build in any way, you are allowed to submit the commit that addresses them before results announcements.

Also you may find these suggestions useful:
https://t.me/tgx_dev/1568
https://t.me/tgx_dev/1561

Reply:
Some Android Studio issue with formatting, probably... 

I forgeted to text what things was completed.

1) Reaction display for bubble/unbubble mode. Bubble mode partly broken, it has wrong positions sometime, but i hate bubble (I'm sure you understand me) mode so i didn't fixed it.
2) Ability to pass animation on message: add existed reaction by tapping on reaction in message or non-existed from context menu. But without most of animations.
2.1) Also, as you say https://t.me/tgx_dev/734 here. I display tooltip if tdlib return error while passing reaction.
3) User can see passed reactions, it partly completed... User can see it only for outgoing messages